File: wikitext-helpers/lyrics_stanza.lua
io.write("; Release Date: xxxx-xx-xx\n\n== Lyrics ==\n")

local buffer = {}

local function out_stanza()
  if #buffer > 0 then
    io.write(table.concat(buffer, "<br/>\n"),"\n\n")
    buffer = {}
  end
end

local function extract_title(line) -- convert <12. Title> to <=== Title ===>
  local _, len = line:find("^%d+%. +")
  if len ~= nil then
    return io.write("=== ",line:sub(len+1)," ===\n")
  end
end

for line in io.lines() do
  line = line:gsub("^%[(.-)%]$","[''%1'']") -- convert [xx] to [''xx'']
  if #buffer == 0 and extract_title(line) then
    -- nothing else to do
  elseif #line > 0 then 
    table.insert(buffer, line)
  else 
    out_stanza()
  end
end

out_stanza() -- take care of any buffered lines
io.write("\n[[Category:Albums (Extreme Metal)]]\n")

File: wikitext-helpers/lyrics_stanza.pl
use v5.36;
say "; Release Date: xxxx-xx-xx\n\n== Lyrics ==";

my @buffer;  # holds the lines we accumulate

sub out_buffer() {
  say join("<br/>\n", @buffer), "\n" if @buffer;
  @buffer = ();
}

while(<>) {
  chomp;
  s/^\[(.*?)\]$/[''$1'']/;  # convert [xxxx] to [''xxx'']
  if(!@buffer and m{^\d+\.\s+}) {  # is it a song title?
    say "=== $' ===";
    next;
  }
  
  length ? push @buffer, $_ : out_buffer();
}
out_buffer(); # take care of any buffered lines
say "\n[[Category:Albums (Extreme Metal)]]";
